pysoundcomparisons/__main__.py

```diff
@@ -152,31 +152,35 @@ def downloadSoundFiles(args, out_path=os.path.join(os.getcwd(), "sound"), db_nee
     db_needed = False if all items can be calculated as keys of catalog.json like FilePathPart {+ WordID}
     """
 
+    from cdstarcat import OBJID_PATTERN
+
     api = _api(args)
     if db_needed:
         db = _db(args)
 
-    catalog_filepath = api.repos.joinpath(
-        'soundfiles', 'catalog.json')
-    if catalog_filepath.exists():
-        catalog_items = jsonlib.load(catalog_filepath)
-    else:
-        print("catalog.json at {} not found.".format(catalog_filepath))
+    catalog = MediaCatalog(api.repos.joinpath('soundfiles', 'catalog.json'))
+    try:
+        catalog[0]
+    except:
+        print("catalog.json at {} is empty.".format(
+            api.repos.joinpath('soundfiles', 'catalog.json')
+        ))
         return
 
     # holds all desired FilePathParts+WordIDs
     desired_keys = set()
 
     # get desired extensions
-    valid_ext = ['mp3', 'ogg', 'wav']
+    valid_ext = catalog.sound_extensions()
     desired_ext = list(set(args.args) & set(valid_ext))
     if len(desired_ext) == 0:
         desired_ext = list(valid_ext)
     else:
         # remove ext from args.args
         args.args = list(set(args.args)-set(valid_ext))
 
-    all_catalog_keys = catalog_items.keys()
+    file_path_uid_map = catalog.file_path_uid_map()
```

Comment:
If `file_path_uid_map` is a cached property, there would be no need to assign it to a local variable.